olive/engine/engine.py

```diff
@@ -848,11 +877,20 @@ def _run_pass(
                 if self.no_search:
                     raise  # rethrow the exception if no search is performed
 
+        # Do we need get the output_model_id from the output_model.model_path like the following?
+        # if output_model != PRUNED_CONFIG and output_model.model_path:
+        #     # get the updated output_model_id from the output model path
+        #     output_model_id = Path(output_model.model_path).name
+        if not p.is_accelerator_agnostic:
```

Comment:
This line should go between 860 and 861.

Reply:
Otherwise outputs from the same not accelerator agnostic pass will overwrite each other's outputs.